vsphere/internal/virtualdevice/virtual_machine_device_subresource.go

```diff
@@ -1048,3 +1048,42 @@ func PciPassthroughApplyOperation(d *schema.ResourceData, c *govmomi.Client, l o
 	}
 	return applyConfig.VirtualDevice, applyConfig.Spec, nil
 }
+
+// PciPassthroughPostCloneOperation normalizes the PCI passthrough devices
+// on a newly-cloned virtual machine and outputs any necessary device change
+// operations. It also sets the state in advance of the post-create read.
+func PciPassthroughPostCloneOperation(d *schema.ResourceData, c *govmomi.Client, l object.VirtualDeviceList) (object.VirtualDeviceList, []types.BaseVirtualDeviceConfigSpec, error) {
+	old, newValue := d.GetChange("pci_device_id")
+	oldDevIds := old.(*schema.Set)
+	newDevIds := newValue.(*schema.Set)
+
+	delDevs := oldDevIds.Difference(newDevIds)
+	addDevs := newDevIds.Difference(oldDevIds)
+	applyConfig := &pciApplyConfig{
+		Client:        c,
+		ResourceData:  d,
+		Spec:          []types.BaseVirtualDeviceConfigSpec{},
+		VirtualDevice: l,
+	}
+	if addDevs.Len() <= 0 && delDevs.Len() <= 0 {
```

Comment:
nitpick, when dealing with lengths I think it's standard to see `.Len() == 0`, just because it's not possible for a negative number.

Reply:
Thanks! I updated this to the following

```go
	if addDevs.Len() == 0 && delDevs.Len() == 0 {
		return applyConfig.VirtualDevice, applyConfig.Spec, nil
	}
```